dev/bots/test.dart

```diff
@@ -1840,19 +1839,28 @@ Future<void> _runDartTest(String workingDirectory, {
     removeLine: useBuildRunner ? (String line) => line.startsWith('[INFO]') : null,
   );
 
+  final TestFileReporterResults test = parseFileReporter(metricFile); // --file_reporter name
+  try {
+    final File info = fileSystem.file(path.join(flutterRoot, 'error.log'));
+    info.writeAsStringSync(json.encode(test.errors));
+  } on fs.FileSystemException catch (e){
```

Comment:
This is me planing for the worst and just caching a file system exception because in theory it can happen https://api.flutter.dev/flutter/dart-io/File/writeAsStringSync.html; The reason I don't fail this is because if it didn't fail at `dart run test --file-reporter...` it means that the failure only occurs after trying to write to a file, something unrelated to the PR in testing

Reply:
And in the case `dart run test` failed first and later on this failed too the check we will have a red build

Reply:
If we fail writing this file, will the recipe fail later on trying to read it?

Reply:
Good question, I think it should fail

Reply:
then better to fail fast with a very specific error message

Reply:
So you suggest raising an exception in here stating that something failed in creating the `error.log` file?

Reply:
Yes. I don't know if there is value in catching this exception if the recipe will just fail later. Then someone investigating a broken tree might think there is something wrong with the recipe, when really the problem was earlier and we could have left clues as the real location of the issue.

Reply:
Fair enough